### PR TITLE
Generic filter view

### PR DESCRIPTION
**Pull request by rmoldo**
## Description
Made filter view generic for skills or projects, added skills and projects counter on filter buttons and implemented filtering for both skills and projects

## Related
https://trello.com/c/ddsin31H

## Link to demo
https://streamable.com/5gwxp1

## Steps to Test or Reproduce

1. Open app
2. Tap on either skills or projects button
3. Select skills / projects
4. Tap on apply button

### CODE REVIEW

**Comment by rmoldo** [comment]
[iOS - Filter Screen - Make Filter view generic](https://trello.com/c/ddsin31H/114-ios-filter-screen-make-filter-view-generic)

File: Shared/Features/Freaks/View/FilterView.swift [hunk 1]
```diff
@@ -1,23 +1,29 @@
 import SwiftUI
 
-struct SkillsView: View {
-    @ObservedObject var viewModel: FreaksViewModel
+struct FilterView: View {
+    @ObservedObject var viewModel: FilterViewModel
     @Binding var showSkillsView: Bool
+    @Binding var showProjectsView: Bool
```
Comment (MihaiPantiru) [inline Shared/Features/Freaks/View/FilterView.swift:6]: We can use just one boolean to track the view presented/closed state to simplify the view logic.

File: Shared/Features/Freaks/ViewModel/FilterViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        case .project:
+            filterItems = projects
+        }
+
+        selectedItems = preselectedItems
+    }
+
+    func didTap(on item: Filterable) {
+        if let index = selectedItems.firstIndex(where: {
+            $0.id == item.id
```
Comment (MihaiPantiru) [inline Shared/Features/Freaks/ViewModel/FilterViewModel.swift:28]: To be consistent with line 37, we can have this check on the same line as line 27.

**Review by MihaiPantiru — CHANGES_REQUESTED** [review]

**Review by MihaiPantiru — APPROVED** [review]

**Review by DejanuTeodora — APPROVED** [review]

File: Shared/Features/Freaks/ViewModel/FreaksViewModel.swift [hunk 3]
```diff
@@ -23,7 +23,9 @@ final class FreaksViewModel: ObservableObject {
     }
 
     func filterFreaks() {
-        if selectedSkills.isEmpty || selectedProjects.isEmpty {
+        if (selectedSkills.isEmpty || selectedProjects.isEmpty) ||
```
Comment (MihaiPantiru) [inline Shared/Features/Freaks/ViewModel/FreaksViewModel.swift:26]: So many rules here :D Won't it be the same as resetting the freaks list before every filtering?

Comment (rmoldo) [reply Shared/Features/Freaks/ViewModel/FreaksViewModel.swift:26]: Yeah I was thinking the same. Should I remove the check?

**Review by MihaiPantiru — APPROVED** [review]

**Review by DejanuTeodora — APPROVED** [review]

**Pull request merged**